sbin/prepareWorkspace.sh

```diff
@@ -623,15 +623,32 @@ downloadDevkit() {
   fi
 }
 
-# Download all of the dependencies for OpenJDK (Alsa, FreeType etc.)
+downloadBootJdkIfNeeded () {
+  if [[ "${BUILD_CONFIG[JDK_BOOT_DIR]}" == "download" ]]; then
+    # The bootDir is used by downloaders.sh methods; it would be nice to change it in future so the download method is unified
+    bootDir="${BUILD_CONFIG[WORKSPACE_DIR]}/downloaded-boot-jdk-${BUILD_CONFIG[OPENJDK_FEATURE_NUMBER]}"
+    if  [ -e "$bootDir" ] ; then
+      echo "Reusing $bootDir"
+    else
+      source "$SCRIPT_DIR/../build-farm/platform-specific-configurations/downloaders.sh"
```

Comment:
Architecturally I'm not a fan of this. The stuff in the root of this repository and `sbin` is supposed to be fully independent. The bits in `build-farm` is for the adoptium build processes and configures the environment for our setup and wraps around the other stuff. Calling out from `sbin` to `platform-specific-configurations` is therefore not ideal.

I like the idea of the option to specify a boot JDK to download, but I think this implementation will need some refinement first.

Reply:
It should be quite strightforward to move downloaders.sh next to common.sh and refactor them so the `$bootDir` is not missused.

I will elaborate on the `bootDir` as I really do not like it. Do you have any prefferences on `downloaders.sh` location?

Reply:
I'm undecided but marginally leaning towards `sbin` or `sbin/common`. WDYT @andrew-m-leonard ?

Reply:
> Architecturally I'm not a fan of this. The stuff in the root of this repository and `sbin` is supposed to be fully independent. The bits in `build-farm` is for the adoptium build processes and configures the environment for our 

Actually there is an misunderstanding, caused by `bootJdk` variable name clash. I will rework it so it is no longer there.

The  `  if  [ -e "$bootDir" ] ; then` is independent  local check, if boot jdk was already downloaded, so it is not downloaded again.

Reply:
done https://github.com/adoptium/temurin-build/pull/3725/commits/00b599c20d3fee1e035a3f23b1f497b51eb2ebcb

Reply:
> I'm undecided but marginally leaning towards `sbin` or `sbin/common`. WDYT @andrew-m-leonard ?

@sxa yes I agree with you, looking at this again I feel uncomfortable that "sbin" code is calling out to "build-farm", as this could lead code-creep in the future architecturally in the wrong direction.
@judovana We should move the downloaders.sh to sbin/common, which is the common place that both sbin and build-farm pull "common" things from please.

Reply:
Sure, Thanx  a lot, on it!

Reply:
done.  Tested on win x86, el8 aarch64 and f38 x64 aganst jdk8,17 and 21.